### DISCUSSION TITLE
Can I generate new component on local branch

### DISCUSSION THREAD

**Discussion by hpublue** [angular/components#25371] [Q&A]
I am learning Angular now. Question. I had generated new branch locally. And would like to know can I generate new component(s) in my local branch?  Thank you

**Comment by rubenperegrina** [comment]
Hi,

Just go to the root of your project and run `ng g component 'name-of-your-component'`

**Reply by nishchay7pixels** [reply]
Hi, Noob question. But how can i use and test the component I created to changes I did. @rubenperegrina

**Reply by rubenperegrina** [reply]
Hi,
If you only want to render the component:

`<your-template-component></your-template-component>`
or just in you are in Angular 15 or greater
`</your-template-component>`

This is your component selector
```
@Component({
  selector: 'your-template-component',
})
```

If you want to use as a page, you can add a route in your routing file like this:
```
const routes: Routes = [
  { path: 'my-template', component: YourTemplateComponent }
];
```

And for test the component, angular cli generate automatically a .spec file,
in this file you can add your tests for this component and run it using 

> ng test


I hope this is useful for you, i think [this docs](https://angular.io/start) may help you, cheers!
